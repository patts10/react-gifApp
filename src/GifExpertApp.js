import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { Footer } from './components/Footer';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

  const [categories, setCategories] = useState(['Squartle']);

  return (
    <>
    <div className='container mx-auto pb-16'>
      <h2 className='text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate my-8'>Gif's Application</h2>
      <AddCategory setCategories={ setCategories } />
      <hr />

      {
        categories.map( category => (
          <GifGrid 
            key={ category } 
            category={ category }
          />
        ))
      }
    </div>
    <Footer />
    </>
  );
};
