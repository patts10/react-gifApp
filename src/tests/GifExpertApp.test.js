import { GifExpertApp } from "../GifExpertApp"
import  { shallow } from 'enzyme'


describe('tests on <GifExpertApp />', () => { 
  
  test('should return component correctly', () => { 
    
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
   })

   test('should show a category list', () => { 
     
    const categories = ['squartle', 'pokemon' ];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
 })