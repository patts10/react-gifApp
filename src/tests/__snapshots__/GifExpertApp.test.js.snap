// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifExpertApp /> should return component correctly 1`] = `
<Fragment>
  <div
    className="container mx-auto pb-16"
  >
    <h2
      className="text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate my-8"
    >
      Gif's Application
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
  </div>
  <Footer />
</Fragment>
`;

exports[`tests on <GifExpertApp /> should show a category list 1`] = `
<Fragment>
  <div
    className="container mx-auto pb-16"
  >
    <h2
      className="text-2xl font-bold leading-7 text-gray-900 sm:text-3xl sm:truncate my-8"
    >
      Gif's Application
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <GifGrid
      category="squartle"
      key="squartle"
    />
    <GifGrid
      category="pokemon"
      key="pokemon"
    />
  </div>
  <Footer />
</Fragment>
`;
